TumbleBitSetup/RsaKey.cs

```diff
@@ -68,6 +79,18 @@ internal byte[] Decrypt(byte[] encrypted)
             return engine.ProcessBlock(encrypted, 0, encrypted.Length);
         }
 
+        public PoupardSternProof CreatePoupardSternProof(PoupardSternSetup setup)
```

Comment:
From a dev perspective, I can understand why you put this code here.  Clearly the proof is an action that is computed over the private key.  

My question is, if we are adding stuff to RSAkey.cs, which is an important file that is used in many projects, does this make it less likely for our code to be incorporated in bouncyCastle?   In other words: is our code more likely to be merged in bouncyCastle if we keep our additions in a separate file, rather than add it to the main RSAkey.cs file?